homeassistant/components/alarm_control_panel/__init__.py

```diff
@@ -52,13 +59,24 @@ def setup(hass, config):
     def alarm_service_handler(service):
         """Map services to methods on Alarm."""
         target_alarms = component.extract_from_service(service)
-
         code = service.data.get(ATTR_CODE)
 
-        method = SERVICE_TO_METHOD[service.service]
```

Comment:
You should create different service schemas for the different services, since your new services have required arguments. The schemas will validate the parameters/attributes and will error before calling the method if a wrong parameter is included.

Include the schemas in the dictionary mapping service to method, like in the cover component. This map will also make sure that only the method connected to the service is called. Look up the dictionary for the correct schema when registering the services.